sync_check/util/chunk.go

```diff
@@ -0,0 +1,313 @@
+package util
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/siddontang/go-mysql/schema"
+)
+
+// chunkRange represents chunk range
+type chunkRange struct {
+	begin interface{}
+	end   interface{}
+	// for example:
+	// containB and containE is true, means [begin, end]
+	// containB is true, containE is false, means [begin, end)
+	containB bool
+	containE bool
+	notNil   bool
+}
+
+// dumpJob is the struct for job of dump
+type dumpJob struct {
+	dbName string
+	table  string
+	column *schema.TableColumn
+	where  string
+	chunk  chunkRange
+}
+
+// newChunkRange return a range struct
+func newChunkRange(begin, end interface{}, containB, containE bool, notNil bool) chunkRange {
+	return chunkRange{
+		begin:    begin,
+		end:      end,
+		containE: containE,
+		containB: containB,
+		notNil:   notNil,
+	}
+}
+
+func getChunksForTable(db *sql.DB, dbname, tableName string, column *schema.TableColumn, timeRange string, chunkSize, sample int) ([]chunkRange, error) {
+	noChunks := []chunkRange{{}}
+	if column == nil {
+		log.Warnf("No suitable index found for %s.%s", dbname, tableName)
+		return noChunks, nil
+	}
+
+	field := column.Name
+
+	// fetch min, max
+	query := fmt.Sprintf("SELECT %s MIN(`%s`) as MIN, MAX(`%s`) as MAX FROM `%s`.`%s` where %s",
+		"/*!40001 SQL_NO_CACHE */", field, field, dbname, tableName, timeRange)
+	log.Debugf("[dumper] get max min query sql: %s", query)
+
+	// get the chunk count
+	cnt, err := GetCount(db, dbname, tableName, timeRange)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if cnt == 0 {
+		log.Infof("no data found in %s.%s", dbname, tableName)
+		return noChunks, nil
+	}
+
+	chunkCnt := cnt / int64(chunkSize)
+	if sample != 100 {
+		// create more chunk for sampling check
+		chunkCnt *= 10
```

Comment:
if sample != 100 means use sampling check, and can check more fragmented by split to more chunk.

Reply:
Please commit it in the code